File: GIBDD.Server/Models/Inspection.cs
using System.ComponentModel.DataAnnotations;

namespace GIBDD.Server.Models
{
    public class Inspection
    {
        [Key]
        public int InspectionId { get; set; }
        
        public DateTime InspectionDate { get; set; }
        public string Result { get; set; } = null!;
        
        public int CarId { get; set; }
        public Car Car { get; set; } = null!;
        
        public int OwnerId { get; set; }
        public Owner Owner { get; set; } = null!;
        
        public int OfficerId { get; set; }
        public Officer Officer { get; set; } = null!;
    }
}

File: gibdd.client/src/app/models/Inspection.ts

export interface AddInspection {
  carId: number;
  ownerId: number;
  officerId: number;
  inspectionDate: string;
  result: string;
}

export interface Inspection {
  inspectionId?: number; // Необязательный для новых записей
  inspectionDate: string; // Формат ISO (например, "2024-12-03T12:00:00Z")
  result: string;
  carId: number;
  officerId: number;
  ownerId: number;
}
